### PR TITLE
Date field does not update value when modified in QField

### PR DESCRIPTION
**Pull request by suricactus**
I am not sure why was that `label.text` was not updating when `main.value` was updated, but this seems to solve the issue. Also note that in my understanding the `main.currentValue` was no longer used in the code, so I took use of it.

Finally, I also changed the `if` on line 45, so when the value is set to `null` it does not become automatically 1970-01-01 .

Steps to reproduce (before this patch is applied): 1) Open a feature form with date field
2) Click on set to now button
3) Delete the contents
4) Click on set to now button
5) Nothing happens

Inspured by #4803 report.

### CODE REVIEW

File: src/qml/editorwidgets/DateTime.qml [hunk 1]
```diff
@@ -36,7 +36,42 @@ EditorWidgetBase {
 
   property bool isDateTimeType: field.isDateOrTime
   property bool fieldIsDate: LayerUtils.fieldType( field ) === 'QDate'
-  property var currentValue: isDateTimeType ? value : Qt.formatDateTime(value, config['field_format'])
+  property var currentValue: {
+    label.text = formatDateTime( value );
```
Comment (nirvn) [inline src/qml/editorwidgets/DateTime.qml:40]: Shouldn't the var be set to fornafDateTime, not the result of setting label.text? It'd in theory create a loop

**Review by nirvn — APPROVED** [review]
@suricactus , I've tested the PR with sample bees. I'm not able to replicate the issue there, but if you are and you say this fixes the situation, it's good to merge.

(I've pushed a commit to fix the currentValue property value and move some null to undefined)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 0d22fa5073dff23b1df9e63b3b422eaedcd6da54: [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4804-63fc34eebc9c612e47c817bc3a98663a8da22a04-arm64-android.apk)
<details>
<summary>Other architectures</summary>

- [arm-neon-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4804-63fc34eebc9c612e47c817bc3a98663a8da22a04-arm-neon-android.apk)
- [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4804-63fc34eebc9c612e47c817bc3a98663a8da22a04-x64-android.apk)
- [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4804-63fc34eebc9c612e47c817bc3a98663a8da22a04-x86-android.apk)
- [**all access arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4804-63fc34eebc9c612e47c817bc3a98663a8da22a04-arm64-android.apk)
- [all access arm-neon-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4804-63fc34eebc9c612e47c817bc3a98663a8da22a04-arm-neon-android.apk)
- [all access x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4804-63fc34eebc9c612e47c817bc3a98663a8da22a04-x64-android.apk)
- [all access x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4804-63fc34eebc9c612e47c817bc3a98663a8da22a04-x86-android.apk)
</details>

**Comment by suricactus** [comment]
Btw, the `null` to `undefined` was done on purpose. This returns true: `null == null` and `null == undefined`, but `null === undefined` returns false. In my experience this sets the widget to 1970-01-01. Is this a desired?

File: src/qml/editorwidgets/DateTime.qml (hunk 1 above)
Comment (suricactus) [reply src/qml/editorwidgets/DateTime.qml]: Actually this is the fix that made it work for me, see:

| before | after |
|---|---|
| Recording 2023-11-28 at 11 29 19 | Recording 2023-11-28 at 11 22 24 |

There is indede a loop though.

**Comment by suricactus** [comment]
@nirvn how shall we proceed here?

**Review by nirvn — APPROVED** [review]
@suricactus , let's merge it but not backport to 3.1?

**Pull request merged**